src/Connection.hs

```diff
@@ -24,16 +27,26 @@ connectWebsockets = do
         shunt (Right e) = writeBChan (st^.csResources.crEventQueue) (WSEvent e)
         runWS = WS.mmWithWebSocket (st^.csResources.crSession) shunt $ \ws -> do
                   writeBChan (st^.csResources.crEventQueue) WebsocketConnect
-                  processWebsocketActions st ws 1
+                  processWebsocketActions st ws 1 zeroTime
     void $ forkIO $ runWS `catch` handleTimeout 1 st
                           `catch` handleError 5 st
+  where
+    zeroTime = UTCTime (ModifiedJulianDay 0) (secondsToDiffTime 0)
 
-processWebsocketActions :: ChatState -> WS.MMWebSocket -> Int64 -> IO ()
-processWebsocketActions st ws s = forever $ do
+processWebsocketActions :: ChatState -> WS.MMWebSocket -> Int64 -> UTCTime -> IO ()
+processWebsocketActions st ws s lastUserTypingNotifTime = do
   action <- STM.atomically $ STM.readTChan (st^.csWebsocketActionChan)
-  WS.mmSendWSAction (st^.csResources.crConn) ws $ convert action
+  if (shouldSendAction action)
+    then do
+      WS.mmSendWSAction (st^.csResources.crConn) ws $ convert action
+      processWebsocketActions st ws (s + 1) =<< getCurrentTime
+    else do
+      processWebsocketActions st ws s lastUserTypingNotifTime
   where
-    convert (UserTyping cId pId) = WS.UserTyping s cId pId
+    convert (UserTyping _ cId pId) = WS.UserTyping s cId pId
+
+    shouldSendAction (UserTyping ts _ _) =
```

Comment:
This doesn't appear to track the channel ID associated with the last typing notification time, so it means that if I type in channel A and switch to channel B in less than two seconds and start typing, the typing in channel B will not cause a server notification of typing to occur.

Reply:
I'll fix this.

Reply:
Thanks!